### PR TITLE
AutoLayout is now a Panel

### PR DESCRIPTION
**Pull request by carldebilly**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] [Docs](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc) have been added/updated
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+{
+	protected override Size ArrangeOverride(Size finalSize)
+	{
+		// Local cache of DependencyProperties
+		var children = Children;
+		var orientation = Orientation;
+		var justify = Justify;
+		var primaryAxisAlignment = PrimaryAxisAlignment;
+		var padding = Padding;
+		var spacing = Spacing is > double.NegativeInfinity and < double.PositiveInfinity and { } s ? s : 0d;
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:25]: what about `var spacing = Spacing.FiniteOrDefault(defaultValue: 0d)` ?

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+		var isHorizontal = orientation == Orientation.Horizontal;
+		var borderThickness = BorderThickness;
+
+		if (BorderBrush is null)
+		{
+			// BorderBrush is not set, so we don't need to consider the border thickness
+			borderThickness = default;
+		}
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:33]:
```suggestion
        // BorderBrush is not set, so we don't need to consider the border thickness
		var borderThickness = BorderBrush == null ? default : BorderThickness
```

File: src/Uno.Toolkit.RuntimeTests/Tests/AutoLayoutTest.cs [hunk 3]
```diff
@@ -181,18 +185,18 @@ public async Task When_AbsolutePosition_WithPadding(bool isStretch, Orientation
...
+	//[DataRow(true, Orientation.Vertical, AutoLayoutAlignment.End, AutoLayoutAlignment.Center, 100, 25)]
+	//[DataRow(true, Orientation.Vertical, AutoLayoutAlignment.Start, AutoLayoutAlignment.Start, 100, 100)]
+	//[DataRow(true, Orientation.Vertical, AutoLayoutAlignment.Center, AutoLayoutAlignment.Start, 100, 100)]
+	//[DataRow(true, Orientation.Vertical, AutoLayoutAlignment.End, AutoLayoutAlignment.Start, 100, 100)]
+	//[DataRow(true, Orientation.Horizontal, AutoLayoutAlignment.Start, AutoLayoutAlignment.Center, 25, 100)]
+	//[DataRow(true, Orientation.Horizontal, AutoLayoutAlignment.Center, AutoLayoutAlignment.Center, 25, 100)]
+	//[DataRow(true, Orientation.Horizontal, AutoLayoutAlignment.End, AutoLayoutAlignment.Center, 25, 100)]
+	//[DataRow(true, Orientation.Horizontal, AutoLayoutAlignment.Start, AutoLayoutAlignment.Start, 100, 100)]
+	//[DataRow(true, Orientation.Horizontal, AutoLayoutAlignment.Center, AutoLayoutAlignment.Start, 100, 100)]
+	//[DataRow(true, Orientation.Horizontal, AutoLayoutAlignment.End, AutoLayoutAlignment.Start, 100, 100)]
```
[the remark this answers is not in the document]
Comment (carldebilly) [reply src/Uno.Toolkit.RuntimeTests/Tests/AutoLayoutTest.cs:199]: They are still unresolved regressions - that's also why this PR is still draft.

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs (hunk 1 above)
Comment (carldebilly) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:25]: We're not familiar with existing extensions in the project... but yeah, looks like it's doing the same job.

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+	{
+		var resultSize = new Size();
+		var anyIndependent = false;
+
+		// Actual available size for independent children is the available size minus the border thickness
+		availableSize = new Size(
+			availableSize.Width - (borderThickness.Left + borderThickness.Right),
+			availableSize.Height - (borderThickness.Top + borderThickness.Bottom));
+
+		for (var i = 0; i < _calculatedChildren!.Length; i++)
```
Comment (jeromelaban) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:361]: Should we check that `_calculatedChildren` can be null?

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**